requirements.txt

```diff
@@ -1,4 +1,3 @@
-Flask==1.0.2
-Werkzeug==0.14.1
-pyOpenSSL==18.0.0
-dnslib==0.9.7
+Flask==1.1.1
+pyOpenSSL==19.0.0
+dnslib==0.9.10
```

Comment:
We should explicitly list all of the dependencies and versions here. Otherwise repeatable builds won't be possible.

Reply:
I've added all Python requirements explicitly.

There are some things which are (probably) not reproducable:
 * the docker base images  (`python:3.6-slim-stretch` and `golang:1.13-stretch`) are regularly updated as long as they are supported;
 * the Pebble build is probably not completely reproducable (the compiler can change due to `golang:1.13-stretch` changing, and not all requirements are vendored and might change).

Reply:
This gets us as close as we reasonably can without pinning to a specific base image SHA and should avoid the most likely issues caused by newly released dependencies.